pkg/operator/ceph/controller/spec.go

```diff
@@ -615,6 +634,51 @@ func StoredLogAndCrashVolumeMount(varLogCephDir, varLibCephCrashDir string) []v1
 	}
 }
 
+// GenerateMDSLivenessProbeExecDaemon generates a liveness probe that makes sure mds daemon is present in fs map,
+// that it can be called, and that it returns 0
+func GenerateMDSLivenessProbeExecDaemon(daemonName string) *v1.Probe {
+	daemonName = strings.TrimLeft(daemonName, "rook-ceph-mds-")
+	var mdsLivenessProbeCmd = []string{
+		"timeout",
+		fmt.Sprintf("%d", mdsCmdTimeout),
+		"sh",
+		"-c",
+		fmt.Sprintf(`outp="$(ceph fs dump --mon-host=$(ROOK_CEPH_MON_HOST)
+		--mon-initial-members=$(ROOK_CEPH_MON_INITIAL_MEMBERS) --format json 2>&1)"
+rc=$?
+if [ $rc -ne 0 ]; then
+	echo "ceph mds dump check failed with the following output:"
+	echo "$outp" | sed -e 's/^/> /g'
+	echo "we couldn't get the information to reliably tell if the MDS should be restarted or not"
+else
+	# get the active and standby mds in the fs map
+	activeMds=$(echo $outp | jq ".filesystems[0].mdsmap.info|map(.name)")
```

Comment:
`.filesystems[0]` assumes there is only 1 filesystem. There may be many, or there could be zero depending on if the last fs is being deleted.

activeMds=$(echo $outp | jq ".filesystems[0].mdsmap.info|map(.name)")

Reply:
I see so should we iterate over every filesystem and get all the common mds?

or

checking the first filesystem would list all the mds?